RevenueCatUI/Views/RevenueCatDeeplinkHandler.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  RevenueCatDeeplinkHandler.swift
+//
+//  Created by Andrés Boedo on 9/17/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable, message: "RevenueCatUI does not support macOS yet")
+@available(tvOS, unavailable, message: "RevenueCatUI does not support tvOS yet")
+struct RevenueCatDeeplinkHandlerView<Content: View>: View {
+    @State private var isShowingPaywall: Bool = false
+    @State private var offeringID: String?
+
+    let content: Content
+
+    init(@ViewBuilder content: () -> Content) {
+        self.content = content()
+    }
+
+    var body: some View {
+        content
+            .onOpenURL { url in
```

Comment:
Not sure if we would like to also support UIKit apps... but I guess that can come in a separate PR.

Reply:
yeah, I figured for UIKit we'd basically do the appDelegate + present from root view controller path

Reply:
Yeah, I think just adding a method that can be called by a developer in the app delegate method would work 👍